packages/apps/comment-nuke/src/nuke.ts

```diff
@@ -1,13 +1,128 @@
-import { Comment, Devvit } from '@devvit/public-api';
-
+import { Comment, Devvit, Post } from '@devvit/public-api';
+ 
 export type NukeProps = {
   remove: boolean;
   lock: boolean;
   skipDistinguished: boolean;
   commentId: string;
   subredditId: string;
 };
-
+ 
+export type NukePostProps = {
+    remove: boolean;
+    lock: boolean;
+    skipDistinguished: boolean;
+    postId: string;
+    subredditId: string;
+  };
+ 
+   async function getAllCommentsInPost(
+    post: Post,
+    result: Comment[] = []
+  ): Promise<Comment[]> {
+    const comments = await post.comments.all();
+    if (comments.length > 0) {
+      for (const comment of comments) {
+        await getAllCommentsInThread(comment, result);
+      }
+    }
+    return result;
+  }
+  
+  export async function handleNukePost(props: NukePostProps, context: Devvit.Context) {
+    const startTime = new Date();
+    let success = true;
+    let message: string;
+  
+    const shouldLock = props.lock;
+    const shouldRemove = props.remove;
+    const skipDistinguished = props.skipDistinguished;
+  
+    if (!shouldLock && !shouldRemove) {
+      return { message: 'You must select either lock or remove.', success: false };
+    }
+  
+    try {
+      const post = await context.reddit.getPostById(props.postId);
+      const user = await context.reddit.getCurrentUser();
+      const modPermissions = await user.getModPermissionsForSubreddit(post.subredditName);
+      const canManagePosts = modPermissions.includes('all') || modPermissions.includes('posts');
+ 
+      console.log(
+        `Mod Info: r/${post.subredditName} u/${user.username} permissions:${modPermissions}: ${
+          canManagePosts ? 'Can mod' : 'Cannot mod'
+        }`
+      );
+  
+      if (!canManagePosts) {
+        console.info('A user without the correct mod permissions tried to nuke all comments of a post.');
+        return {
+          message: 'You do you not have the correct mod permissions to do this.',
+          success: false,
+        };
+      }
+  
+      const allComments = await getAllCommentsInPost(post);
+      const comments = allComments.filter(
+        (eachComment: any) => !(skipDistinguished && eachComment.isDistinguished())
+      );
+  
+      const lockPromises: Promise<any>[] = [];
+      const removePromises: Promise<any>[] = [];
+  
+      for (const eachComment of comments) {
+        const commentId = eachComment.id;
+        if (shouldLock && !eachComment.locked) {
+          lockPromises.push(eachComment.lock());
+        }
+  
+        if (shouldRemove && !eachComment.removed) {
```

Comment:
if we only allow removal XOR locking, I think we can move this conditional outside of the loop. something like:

```ts
if (shouldLock) await Promise.all(comments.map(comment => comment.locked || comment.lock()))
if (shouldRemove) await Promise.all(comments.map(comment => comment.removed || comment.remove()))
```

smart thinking on checked locked / removed states. I'm not sure why those aren't considered by the API internally before making an expensive call.